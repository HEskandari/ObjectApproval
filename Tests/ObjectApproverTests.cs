using System.Collections.Generic;
using ApprovalTests.Reporters;
using ObjectApproval;
using Xunit;

[assembly: UseReporter(typeof(ClipboardReporter), typeof(DiffReporter))]

public class ObjectApproverTests
{
    [Fact]
    public void Example()
    {
        var person = new Person
            {
                Title = Title.Mr,
                GivenNames = "John",
                FamilyName = "Smith",
                Spouse = "Jill",
                Childres = new List<string> {"Sam", "Mary"},
                Address = new Address
                    {
                        Street = "1 Puddle Lane",
                        Country = "USA"
                    }
            };

        ObjectApprover.VerifyWithJson(person);
    }

    class Person
    {
        public string GivenNames;
        public string FamilyName;
        public string Spouse;
        public Address Address;
        public List<string> Childres;
        public Title Title;
    }

    class Address
    {
        public string Street;
        public string Suburb;
        public string Country;
    }
}

public enum Title
{
    Mr
}