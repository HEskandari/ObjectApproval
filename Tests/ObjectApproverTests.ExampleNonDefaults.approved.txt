{
  "$type": "ObjectApproverTests.Person",
  "GivenNames": "John",
  "FamilyName": "Smith",
  "Spouse": "Jill",
  "Address": {
    "$type": "ObjectApproverTests.Address",
    "Street": "1 Puddle Street",
    "Country": "USA"
  },
  "Children": {
    "$type": "List<string>",
    "$values": [
      "Sam",
      "Mary"
    ]
  },
  "Dob": "9999-12-31T23:59:59.9999999",
  "Id": "ebced679-45d3-4653-8791-3d969c4a986c"
}